manager/src/main/java/com/redhat/service/bridge/manager/dao/WorkDAO.java

```diff
@@ -24,7 +24,10 @@ public Work findByManagedResource(ManagedResource managedResource) {
         return list("#Work.findByWorkerId", Parameters.with("workerId", workerId));
     }
 
-    public int rebalanceWork(String workerId, ZonedDateTime lastUpdated) {
-        return update("#Work.updateWorkerId", Parameters.with("workerId", workerId).and("age", lastUpdated));
+    public int reconnectDroppedWorkers(String workerId, ZonedDateTime lastUpdated) {
+        return update("#Work.reconnectDroppedWorkers",
+                Parameters.with("workerId", workerId)
+                        .and("now", ZonedDateTime.now())
```

Comment:
I pass "now" from the Java code vs using the DB's "now" as there's a chance they're in different timezones/configured differently.. and since the original "submitted" date comes from Java I thought this safest.

Reply:
So just to confirm: From a users perspective, the timeout on a resource provisioning is now dependent on how many workers get involved. So rather than it being absolute e.g. if 10 minutes elapses from the point where you submit and we haven't deployed, that's a fail, we're now saying that we'll give each worker at most "timeout" minutes to do the work.

I've followed correctly?

Reply:
Yes.

This is the scenario I worry about...

- WorkA on WorkerY submitted at X.
- WorkerY fails (Pod dies, whatever). WorkA is stalled.
- Our "liveliness for Work" runs every 5m.
- Assuming WorkerZ restarts instantly then WorkA could be completed X+5m (IDK if the scheduler runs immediately or after the first 5m or at set 5m intervals past the hour etc.. but I hope you get the idea).- 
- If the timeout was 3m then WorkerZ will timeout WorkA immediately and mark as FAILED (as it first processes the work at 5m after submittedAt exceeding the timeout).
- If the timeout is longer than our "liveliness for Work", e.g. 6m, WorkerZ gets 1m to complete before FAILING the Work.
- The issue compounds itself if for whatever reason a Worker keeps dying.

Resetting "submittedAt" means:
- Each Worker will get the full timeout period.
- We loose the "total" time (from User saying "create this Bridge") unless we add an "originalSubmittedAt" immutable field.

Reply:
@manstis You're resetting `submittedAt` on the `Work` item and not the `ManagedResource`. So we always have the total time to provision a resource by doing `managedResource.getPublishedAt() - managedResource.getSubmittedAt()` unless I'm missing something?

Reply:
Great, you are correct I'm only looking at Work.

 I notice we do not set a Processor's published date anywhere.

It's not in the sending `BridgeExecutorController` or receiving `ShardBridgesSyncAPI` or supporting `ProcessorServiceImpl#updateProcessor(..)`.

I'd have expected the Shard's call back to the Control Plane to do it (similarly to where they delete the applicable record when de-provisioing a Processor)... it makes sense to me to set it there. Would you like me to do it as part of this PR? I can check Bridges too.. but I'm working on those next for MGDBOR-363 so can cover them there.

Reply:
Yes please @manstis . And then I think we're good to go 🔥

Reply:
@robpblake All done, all green.. it will make the Sprint!!